core/src/main/java/org/apache/iceberg/io/DataWriter.java

```diff
@@ -37,16 +38,23 @@
   private final PartitionSpec spec;
   private final StructLike partition;
   private final ByteBuffer keyMetadata;
+  private final SortOrder sortOrder;
   private DataFile dataFile = null;
 
   public DataWriter(FileAppender<T> appender, FileFormat format, String location,
                     PartitionSpec spec, StructLike partition, EncryptionKeyMetadata keyMetadata) {
+    this(appender, format, location, spec, partition, keyMetadata, null);
+  }
+
+  public DataWriter(FileAppender<T> appender, FileFormat format, String location,
```

Comment:
We instantiate `DataWriter` in `FileAppenderFactory` implementations. It is absolutely correct that we don't propagate the sort order id now. However, should we consider adding a comment that explains why? Like we cannot guarantee the incoming records are sorted as needed?

Reply:
Sorry to make sure I understand the suggestion correctly, do you mean to add a comment here to discourage people from using it for now?

Reply:
I was referring to the place where, for example, `SparkAppenderFactory` instantiates `DataWriter`:

```
  @Override
  public DataWriter<InternalRow> newDataWriter(EncryptedOutputFile file, FileFormat format, StructLike partition) {
    return new DataWriter<>(newAppender(file.encryptingOutputFile(), format), format,
        file.encryptingOutputFile().location(), spec, partition, file.keyMetadata());
  }
```

Here we are using the constructor without sort order. I thought about a comment explaining why but I think we should do that later. We will eventually add sort order to classes like `SparkAppenderFactory` and then can add the comment.

To sum up, ignore my original comment.